Training/Training/Training.c

```diff
@@ -1,14 +1,194 @@
-// ------------------------------------------------------------------------------------------------
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// main.c
+// Program to check for palindrome of a string or an integer
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+
+#include<ctype.h>
+#include<errno.h>
+#include <limits.h>
+#include <malloc.h>
+#include<stdio.h>
+#include <stdlib.h>
+#include<string.h>
+
+// Input limit is 200 characters
+#define MAX 202
+#define PALINDROME 1
+#define NOTPALINDROME 0
+#define INVALID -1
+#define ANSI_COLOR_RED     "\x1b[31m"
+#define ANSI_COLOR_GREEN   "\x1b[32m"
+#define ANSI_COLOR_RESET   "\x1b[0m"
+#define ANSI_COLOR_YELLOW  "\x1b[33m"
+
+/// <summary>
+/// Function which checks if a given string is palindrome or not
+/// </summary>
+int PalindromeChecker (char* phrase) {
+   int i = 0, j = 0, length, newLength, flag = 0;
+   char temp[MAX];
+   length = (int)strlen (phrase);
+   if (length > MAX - 2) {
+      printf ("\nLimit exceeded");
+      return INVALID;
+   }
+   //Eliminating special characters by moving the remaining character into a temporary array
+   while (i < length) {
+      if (isalnum (phrase[i]) != 0)
+         temp[j++] = phrase[i];
+      i++;
+   }
+   temp[j] = '\0';
+   newLength = (int)strlen (temp);
+   //Checking invalid text
+   if (newLength == 0)
+      return INVALID;
+   i = 0;
+   while (i < newLength / 2) {
+      if (toupper (temp[i]) == toupper (temp[newLength - 1 - i]))
+         flag++;

```

Comment:
Here you are checking if all letters till half of the length are equal instead you can check for not equal so this expression can be simplified.